File: leetcode3/황은지/3838. Weighted Word Mapping.js
/**
 * @param {string[]} words
 * @param {number[]} weights
 * @return {string}
 */
var mapWordWeights = function(words, weights) {
    let result=""
    for(const word of words){
        let sum=0;
        for(let i=0;i<word.length;i++){
            const char=word.charCodeAt(i);
            sum+=weights[char-"a".charCodeAt(0)];
        }
        result+=String.fromCharCode("z".charCodeAt(0)-(sum%26));
    }

    return result
};

File: leetcode3/남효정/3838. Weighted Word Mapping.py
class Solution:
    def mapWordWeights(self, words: List[str], weights: List[int]) -> str:
        ans = []

        # ord()의 합과 알파벳 거꾸로 정렬한 순서의 합이 항상 122인 것을 이용
        for word in words:
            sum_score = 0
            for char in word:
                sum_score += weights[ord(char)-97]
            ans.append(chr(122 - sum_score % 26))
        
        return ''.join(ans)